src/book/02-system/02-subsystems/01-input.mdx

```diff
@@ -24,8 +29,959 @@ The cameras have the following parameters that are used by the object detection
+| `Width`                      | Integer     | Width of the source image                                                                                    |
+| `Height`                     | Integer     | Height of the source image                                                                                   |
+| `OffsetX`                    | Integer     | Horizontal offset from the origin to the ROI (in pixels)                                                     |
+| `OffsetY`                    | Integer     | Vertical offset from the origin to the ROI (in pixels)                                                       |
+| `ExposureAuto`               | Enumeration | Sets the automatic exposure mode                                                                             |
+| `GainAuto`                   | Enumeration | Sets the automatic gain mode. In automatic modes, the camera adjusts the gain to maximize the dynamic range. |
+| `BalanceWhiteAuto`           | Enumeration | Compensates for colour shifts caused by different lighting conditions.                                       |
+| `AcquisitionFrameRateEnable` | Boolean     | Allows manual control of the frame rate                                                                      |
+| `AcquisitionFrameRate`       | Float       | User controlled acquisition frame rate in hertz                                                              |
+
+These parameters are set for each camera as **configuration values** in the [Camera module](https://github.com/NUbots/NUbots/tree/main/module/input/Camera/data/config). Configuration files stored in the [config/Cameras](https://github.com/NUbots/NUbots/tree/main/module/input/Camera/data/config/Cameras) directory are common to all robots, and configs that are specific to a particular robot are located in that robot's respective directory. The values for the left camera on the robot will be stored in `Left.yaml`. The values for the right camera on the robot will be stored in `Right.yaml`.
 
 The parameters are used in the **Camera module** to find and set up the cameras. The Camera module emits [Image messages](https://github.com/NUbots/NUbots/tree/main/shared/message/input/Image.proto).
 
-The [projection tool](https://github.com/NUbots/NUbots/blob/main/shared/utility/vision/projection.hpp), based on [panotools' fisheye projection calculations](https://wiki.panotools.org/Fisheye_Projection), maps a portion of the surface of a sphere to a flat image. The type of projection is specified by the above parameter `lens.projection`.
+The [projection tool](https://github.com/NUbots/NUbots/tree/main/shared/utility/vision/projection.h), based on [panotools' fisheye projection calculations](https://wiki.panotools.org/Fisheye_Projection), maps a portion of the surface of a sphere to a flat image. The type of projection is specified by the above parameter `lens.projection`.
```

Comment:
This is here twice? And the link in this version is wrong, it should be hpp still.

Reply:
`https://github.com/NUbots/NUbots/tree/main/shared/utility/vision/projection.h` ????? i thought we renamed all of the `.h` files to `.hpp` in the repo ages ago?

Reply:
Good spot. Not sure how that happened